staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -435,6 +443,9 @@ func (s *DelegatingAuthenticationOptions) getClient() (kubernetes.Interface, err
 	if s.CustomRoundTripperFn != nil {
```

Comment:
this was just added in 1.22... I'm not really sure whether I'd expect this to go before or after the tracing wrappers

Reply:
This gave me an idea for simplifying this PR.  We essentially want to add a custom round tripper function for tracing, so I'm going to use this field for the tracing roundtripper as well.  I changed the behavior of WithCustomRoundTripper to allow invoking it multiple times.  When it is invoked multiple times, it calls the round trippers in order, rather than overwriting the roundtripper.

In this case, the roundtripper for tracing is invoked last.  In general, it shouldn't matter.  But having tracing go last is probably best, as it would allow other wrappers to add tags or baggage before the context is added to the http request.